### PR TITLE
Incorrect conversion between integer types

### PR DESCRIPTION
**Pull request by Ayrtat**
* Add a check if a value is not out of Integer limit

### CODE REVIEW

File: utils/common.go [hunk 1]
```diff
@@ -76,17 +77,18 @@ func MultipleChoiceInputReader(prompt string, choices []string) string {
 		fmt.Printf("You must provide an Integer. You provided %s", temp)
 		fmt.Println()
 		return MultipleChoiceInputReader(prompt, choices)
-	} else {
```
Comment (Ayrtat) [inline utils/common.go:79]: @OleDakotaJoe 
Little refactored the code.
Also I think that there is a lack -- scanning input within `MultipleChoiceInputReader` that does not allow to create tests for the method.

What do you think about to factor `os.Stdin` out? And pass a reader as a method's parameter

**Comment by OleDakotaJoe** [comment]
There is a test in the common_test.go file.
Called " Test_GivenInput_ReturnInput_StringInputReader"
if you look at this test, it tests the "StringInputReader" by passing in a pre-defined []byte of content.

This content is loaded into os.Stdin.
thn the function is called, 
then the content is expected to be the output of the StringInputReader and tested.

Can you add a test to verify that an out of bounds integer will not throw an error?
You should be able to use the same pattern for passing in data to os.Stdin as in the test mentioned.

(I currently have tests for multple choice reader in another test file, but it's more of an end to end test)

**Comment by Ayrtat** [comment]
> There is a test in the common_test.go file. Called " Test_GivenInput_ReturnInput_StringInputReader" if you look at this test, it tests the "StringInputReader" by passing in a pre-defined []byte of content.
> 
> This content is loaded into os.Stdin. thn the function is called, then the content is expected to be the output of the StringInputReader and tested.
> 
> Can you add a test to verify that an out of bounds integer will not throw an error? You should be able to use the same pattern for passing in data to os.Stdin as in the test mentioned.
> 
> (I currently have tests for multiple choice reader in another test file, but it's more of an end to end test)
If you want to me check `MultipleChoiceInputReader`, then this seems impossible. I got the idea how to fake stdin but the problem is that the function is recursive and it requires to input values until it is correct. The test will hang on infinitely recursive invocation of `MultipleChoiceInputReader`. Another option is that I can move the check out of the method and test it as separate

**Comment by OleDakotaJoe** [comment]
Interesting point, didn't think about the recursion, sorry reviewed that code at like midnight.

It might be possible to intercept stdout to ensure the function calls itself again.

I'll look into this another time.
________________________________

**Pull request merged**